### DISCUSSION TITLE
TP-Link Kasa Smart KP303 Plug Power Strip Support

### DISCUSSION THREAD

**Discussion by SlyKittyCat** [jamesmccannon02/OctoPrint-Tplinkautoshutdown#11] [Ideas]
TP-Link Kasa Smart KP303 Plug Power Strip Support
Please and Thank You!

**Comment by escfoe2** [comment]
I second this, but I would like to see the other mentioned features in the description of this plugin come to fruition first.

**Comment by CodingTheUnknown** [comment]
@SlyKittyCat @escfoe2 If you wouldn't mind, please can you test the support for the KP303. I have developed a new version. You will need to uninstall the current TpHandler and then install the plugin using the URL

[https://github.com/jamesmccannon02/OctoPrint-Tplinkautoshutdown/archive/develop-powerStrip.zip](https://github.com/jamesmccannon02/OctoPrint-Tplinkautoshutdown/archive/develop-powerStrip.zip)

**Comment by SlyKittyCat** [comment]
installed it turns off manually but will not turn on manually
it turns off automatically but for all 3 outlets on the strip




@SlyKittyCat @escfoe2 If you wouldn't mind, please can you test the support for the KP303. I have developed a new version. You will need to uninstall the current TpHandler and then install the plugin using the URL
https://github.com/jamesmccannon02/OctoPrint-Tplinkautoshutdown/archive/develop-powerStrip.zip

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.


| 
| 
| 
|  |  |

 |

 |
| 
|  | 
escfoe2 - Overview

escfoe2 has 2 repositories available. Follow their code on GitHub.
 |

 |

 |





| 
| 
| 
|  |  |

 |

 |
| 
|  | 
SlyKittyCat - Overview

GitHub is where SlyKittyCat builds software.
 |

 |

 |

**Reply by CodingTheUnknown** [reply]
Thanks for that can you send me a screenshot of the settings you are using

**Comment by SlyKittyCat** [comment]
attached

    On Wednesday, February 24, 2021, 9:21:56 PM EST, jamesmccannon02 <notifications@github.com> wrote:  
 
 


Thanks for that can you send me a screenshot of the settings you are using

**Reply by CodingTheUnknown** [reply]
Hi, the image hasn't attached

**Comment by SlyKittyCat** [comment]
tp.jpg 

    On Thursday, February 25, 2021, 8:17:41 AM EST, jamesmccannon02 <notifications@github.com> wrote:  
 
 


Hi, the image hasn't attached

**Reply by CodingTheUnknown** [reply]
Sorry still isn't attaching just showing the sub text, are you sending it over email? If you are i think that is the issue as it doesn't prepare the file correctly.

**Comment by SlyKittyCat** [comment]
![tp](https://user-images.githubusercontent.com/49165917/109167232-8acee700-774b-11eb-9c20-9a695a08a58e.JPG)

**Reply by CodingTheUnknown** [reply]
Perfect thanks. When you click update connection, does this then give the details of your smart strip?

**Reply by SlyKittyCat** [reply]
When I click update connection it gives me the 3 sockets under preference but no names for them and no device name or firmware info under device info

**Reply by CodingTheUnknown** [reply]
Ok thanks will look into this. Thanks for the assistance so far

**Reply by SlyKittyCat** [reply]
my pleasure appreciate your work

**Reply by CodingTheUnknown** [reply]
So i believe i have found the cause of the problem... Or at least one of them, so if you wouldn't mind could you uninstall the plugin and then re-install it again the same way. This will then bring down the correct one with the changes made.

- [ ] Believe i solved the auto turn off issue
- [ ] Believe i solved the manual switch button (Currently the manual switch will turn all sockets off but this will be changed in a future version)

**Reply by SlyKittyCat** [reply]
what version is the new one mine shows 1.0.1

**Reply by CodingTheUnknown** [reply]
1.0.1 is the new one. I have made it 1. Because when released it will be a major change

**Reply by SlyKittyCat** [reply]
ok just checking manual on does not function but manual off does turn off everything as you said

**Reply by CodingTheUnknown** [reply]
Ok thanks for that. One step forward I suppose. I’ll go back and take another look. Thanks once again

**Reply by CodingTheUnknown** [reply]
If you can, could you upload the .log for the printer might give a hint

**Reply by RodLaird** [reply]
Hi James, I am not seeing any of the plugs on the strip - certainly nothing like the grab above... It does show the firmware code so there is clearly some communication. 

![image](https://user-images.githubusercontent.com/54025043/109371450-189c0500-78f9-11eb-8ed2-b2d78f78b3b0.png)

**Reply by CodingTheUnknown** [reply]
Hi that’s because your running the public version not the separate one that you need to use the link for. Uninstall that version then go and install the plugin using url and use the following url. That will then show you the a different gui
https://github.com/jamesmccannon02/OctoPrint-Tplinkautoshutdown/archive/develop-powerStrip.zip

**Reply by RodLaird** [reply]
Thanks James - will do at end of current print and report back

**Reply by CodingTheUnknown** [reply]
@RodLaird When you install and setup your smartStrip, please can you try inputting the smart strip ip choosing the smart strip from the dropdown. Then save and restart your octoprint server, then go back into the settings click update connection choose what you want your printer to do on each plug and then save. I think there is a bug within a library i am using that prevents what i am trying to do so i need to figure out if is the problem i believe it is.

**Comment by RodLaird** [comment]
Will do and report
cheers


Rod Laird
+61 418 166 105

**Reply by RodLaird** [reply]
Can manually turn off the priniter - but the auto turn-off does not seem to work. Have enabled debugging and try again (thought I thought I had enabled it for the last print job but se no log. Will restart octo and see what happens. )
![image](https://user-images.githubusercontent.com/54025043/109409015-1ada8e00-79e3-11eb-828f-9b82ad5cbd0f.png)

**Comment by RodLaird** [comment]
Sorry about the delay James; “life intervenes”
 
I have tried a number of short prints with the “turn off” click box on a couple of the strip selections. Never turns anything off… So have attached a log which may should cover a couple of print events. I had enabled debug logging for your plugin…
 
cheers
 
Rod Laird
+61 4 1816 6105
